storage/block/block.go

```diff
@@ -169,6 +180,7 @@ func (b *dbBlock) Seal() {
 	b.writable = false
 	if stream := b.encoder.Stream(); stream != nil {
 		b.segment = stream.Segment()
+		b.checksum = digest.SegmentChecksum(b.segment)
```

Comment:
Do you need to set b.checksum to 0 on Unseal method? We can depend on writable I guess, but seems like an implicit dependency that is hard to remember to keep in future code changes.